src/main/java/eu/europeana/api/recommend/web/RecommendController.java

```diff
@@ -107,10 +107,9 @@ public ResponseEntity acceptSet(
             @Valid @RequestBody String[] ids,
             @RequestHeader(value = HttpHeaders.AUTHORIZATION, required = false) String authToken)
             throws RecommendException {
-        //String apikey = checkCredentials(authToken, wskey);
+        String userId = checkCredentials(authToken, wskey, true);
         validateRecordIds(ids);
-
-        // TODO for now we simply return the same number of recommendations as received
+        recommendService.submitUserSignals(ids, RecommendationConstants.USER_SIGNAL_ACCEPT, userId, authToken, wskey);
         return recommendSet(setId, ids.length, 0, null,  wskey, authToken);
```

Comment:
Either we should use the pageSize parameter here, or we should delete the method parameter. Specification do not say if we need to support pageSize here, so I'm inclinded to remove (but perhaps check with Hugo first?)

Reply:
I will check this with Hugo